Fix AugmentationSequential crash with mix-only augmentations

Saksham-st12 opened the pull request:
#### Summary

This PR fixes a `RuntimeError` in `AugmentationSequential` that occurs when it is initialized with only mix augmentations (e.g. `RandomMixUpV2`, `RandomCutMixV2`) and `random_apply=True`.

#### Issue

In `get_random_forward_sequence`, mix augmentations are excluded from the multinomial sampling step since at most one mix operation is allowed per forward pass. When the augmentation sequence contains only mix augmentations, this results in a zero-sum `multinomial_weights` tensor. Calling `torch.multinomial` on this distribution raises:
RuntimeError: invalid multinomial distribution (sum of probabilities <= 0)

#### Fix

The following changes were made in `kornia/augmentation/container/image.py`:

- Added an explicit guard to detect the zero-sum case and skip multinomial sampling when no non-mix augmentations are available.
- Updated the mix-augmentation injection logic to correctly handle empty indices, ensuring that a single mix augmentation is applied when appropriate.

This preserves the intended behavior of applying at most one mix augmentation per forward pass without causing a crash.

#### Tests

- Added a regression test `test_mixup_cutmix_only` in `tests/augmentation/container/test_augmentation_sequential.py`.
- The test verifies that `AugmentationSequential` initialized with only mix augmentations no longer crashes and produces an output tensor of the expected shape.

Fixes #3300

Several CI checks are failing across collector, tests (multiple Python versions), tutorials, and typecheck.

The failures appear to be repo-wide and unrelated to the changes in AugmentationSequential (e.g. typecheck diagnostics and tutorial/test failures in unrelated modules).

All tests related to this change pass locally.
Please let me know if you’d like me to rebase, adjust scope, or if these failures are known on main.

edgarriba commented on tests/augmentation/container/test_augmentation_sequential.py:
```diff
@@ -59,6 +59,23 @@ def test_mixup(self, inp, random_apply, same_on_batch, device, dtype):
         reproducibility_test(inp, aug)
 
+    def test_mixup_cutmix_only(self, device, dtype):
+        # Regression test for https://github.com/kornia/kornia/issues/3300
```
not a good idea to refer to issues in comments

edgarriba commented on kornia/augmentation/container/image.py:
```diff
@@ -189,18 +189,25 @@ def get_random_forward_sequence(self, with_mix: bool = True) -> Tuple[Iterator[T
-            # enable replacement if non-mix augmentation is less than required
-            replacement=num_samples > multinomial_weights.sum().item(),
-        )
+        if (multinomial_weights.sum() == 0).all():
```
ok, i see. I think it's fine then with sum. The only thing i see is that eventuually sum is done in line 199  too. I force that sum happens only once

Saksham-st12 commented:
Hi, I’m running into an issue with the “verified signature” requirement.
I’ve tried re-signing the latest commit using SSH signing, but GitHub still marks the commit as Unverified and prompts me to upload a signing SSH key.

Could you please confirm whether this repository requires GPG-only signatures, or if SSH-signed commits should be accepted here?
I’m happy to re-sign the commit in the preferred way once clarified.

Saksham-st12 commented:
> Not all the suggestions seem to be applied

I’ve re-checked the latest commit and applied all the suggestions I’m aware of, including consolidating the weight sum computation and updating the zero-weight handling logic.

If there’s any specific suggestion that still appears missing, could you please point it out so I can address it directly?

Saksham-st12 commented:
Hi @andrew-shc,

I’ve addressed the review feedback on this PR, rebased and squashed it to keep the diff minimal, and fixed earlier commit-signing issues.

This PR fixes a crash in AugmentationSequential when only mix augmentations are used and adds a small regression test.

If you have time, I’d really appreciate a quick review of the current state. Thanks!
